google-cloud-logging/src/test/java/com/google/cloud/logging/LoggingTest.java

```diff
@@ -100,6 +104,18 @@ public void testEntryListOption() {
             "folders/test-folder");
   }
 
+  @Test
+  public void testFilterUpdate() {
+    Map<Option.OptionType, ?> options = LoggingImpl.optionMap(EntryListOption.filter(FILTER));
+    assertThat((String) EntryListOption.OptionType.FILTER.get(options)).isEqualTo(FILTER);
+    Map<Option.OptionType, ?> updated = LoggingImpl.updateFilter(options);
+    assertTrue(((String) EntryListOption.OptionType.FILTER.get(updated)).contains("timestamp"));
+    assertFalse(options == updated);
+    assertNotEquals(EntryListOption.OptionType.FILTER.get(updated), FILTER);
+    Map<Option.OptionType, ?> anotherUpdated = LoggingImpl.updateFilter(updated);
+    assertTrue(anotherUpdated == updated);
+  }
+
```

Comment:
Would it be helpful to add a test that does paging like the linked error report?

Reply:
Most of our tests are done with mocks and apparently there is no much point of simulating such error (e.g. this code does not handle the error but rather tries to prevent the problem). I believe this would be better covered by environment tests to make sure paging works as expected